apps/meteor/ee/app/livechat-enterprise/server/hooks/handleNextAgentPreferredEvents.ts

```diff
@@ -18,12 +18,15 @@ const normalizeDefaultAgent = (agent?: Pick<IUser, '_id' | 'username'> | null):
 	return { agentId, username };
 };
 
-const getDefaultAgent = async (username?: string): Promise<SelectedAgent | null> => {
-	if (!username) {
+const getDefaultAgent = async ({ username, id }: { username?: string; id?: string }): Promise<SelectedAgent | null> => {
+	if (!username && !id) {
 		return null;
 	}
 
-	return normalizeDefaultAgent(await Users.findOneOnlineAgentByUserList(username, { projection: { _id: 1, username: 1 } }));
+	if (id) {
+		return normalizeDefaultAgent(await Users.findOneOnlineAgentById(id, { projection: { _id: 1, username: 1 } }));
```

Comment:
These 2 calls should pass the setting value too. The livechat_enabled_when_agent_idle. Not sure if it should be on this PR or in another one 🤔

Reply:
hmmm wdyt about leaving this to another task? This way we can focus on better refine and testing this
I'll create a task

Reply:
Nice, pls create task 🙏🏽